### ISSUE TITLE
Opening telescope live_grep from visual mode (lua) bugged

### ISSUE THREAD

**Issue by devtoi** [nvim-telescope/telescope.nvim#1029]
### Description
Entering the live_grep picker from visual mode enters the picker in wierd state where you have to press escape to enter something and `default_text` isn't applied correctly.

**Expected Behavior**
Picker enters expected state where provided `default_text` is filled in the input box

**Actual Behavior**
Picker enters unexpected state where provided `default_text` is *not* filled in the input box and you have to press Escape to start entering something.

### Details

<details><summary>Reproduce</summary>

1. nvim -nu test.vim
2. Press F7, picker shows up with a text prefilled in the input bux
3. Close picker
4. Visually select some text (enter visual mode)
5. Press F7, picker opens in wierd state with no prefilled text in the input box

</details>

<details><summary>Environment</summary>

- nvim --version output: 
```
NVIM v0.6.0-dev+48-g5377b2b00
Build type: RelWithDebInfo
LuaJIT 2.0.5
Compilation: /usr/bin/cc -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions     -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security     -fstack-clash-protection -fcf-protection -Wp,-U_FORTIFY_SOURCE -Wp,-D_FORTIFY_SOURCE=1 -DNVIM_TS_HAS_SET_MATCH_LIMIT -O2 -g -Og -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL=3 -I/home/toi/.cache/paru/clone/neovim-git/src/build/config -I/home/toi/.cache/paru/clone/neovim-git/src/neovim-git/src -I/usr/include -I/home/toi/.cache/paru/clone/neovim-git/src/build/src/nvim/auto -I/home/toi/.cache/paru/clone/neovim-git/src/build/include
Compiled by toi

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

Run :checkhealth for more info
```
- Operating system: 5.12.15-arch1-1
- Telescope commit: 8c3f2b6

</details>

<details><summary>Configuration</summary>
<p>

```vimL
set nocompatible hidden laststatus=2

if !filereadable('/tmp/plug.vim')
  silent !curl --insecure -fLo /tmp/plug.vim
    \ https://raw.githubusercontent.com/junegunn/vim-plug/master/plug.vim
endif

source /tmp/plug.vim
call plug#begin('/tmp/plugged')
Plug 'nvim-lua/popup.nvim'
Plug 'nvim-lua/plenary.nvim'
Plug 'nvim-telescope/telescope.nvim'
call plug#end()

autocmd VimEnter * PlugClean! | PlugUpdate --sync | close
lua require('telescope').setup()

lua << EOF
function broken_picker()
  local telescope = require "telescope.builtin"
  local opts = {
    search_dirs = { "./" },
    default_text = "this will not show when opened from visual mode",
    initial_mode = "insert",
  }
  telescope.live_grep(opts)
end

vim.api.nvim_set_keymap("v", "<F7>", "<cmd>lua broken_picker()<CR>", { noremap = true } ) -- Doesn't work as expected
vim.api.nvim_set_keymap("n", "<F7>", "<cmd>lua broken_picker()<CR>", { noremap = true } ) -- Works as expected
EOF
```

**Comment by devtoi** [comment]
I found the cause of the bug. I would say it probably is not directly related to telescope or popup but rather the keybinding using `"<cmd>lua broken_picker()<CR>"` instead of `":lua broken_picker()<CR>"`. The later works.
I am not exactly sure if it still kind of a problem though, so feel free to take a second look and close if you think this is not an issue for telescope.

**Comment by Conni2461** [comment]
You should probably read `:help <cmd>`.

TLDR `<cmd>` doesnt change mode and telescope just doesn't work with visual mode because its not designed that way.
